### ISSUE TITLE
Application does not cleanly shut down after SIGINT/SIGTERM (systemd)

### ISSUE THREAD

**Issue by nemec** [timothycrosley/streamdeck-ui#189]
I've installed streamdeck-ui as a userspace systemd service but when I run `systemctl --user stop streamdeck` to stop the service, it is not cleanly shut down. Upon starting the service again, I get the following traceback before core dump:

```
Traceback (most recent call last):
  File "/home/<user>/.local/bin/streamdeck", line 8, in <module>
    sys.exit(start())
  File "/home/<user>/.local/lib/python3.9/site-packages/streamdeck_ui/gui.py", line 634, in start
    items = api.open_decks().items()
  File "/home/<user>/.local/lib/python3.9/site-packages/streamdeck_ui/api.py", line 106, in open_decks
    deck.reset()
  File "/home/<user>/.local/lib/python3.9/site-packages/StreamDeck/Devices/StreamDeckOriginalV2.py", line 111, in reset
    self.device.write_feature(payload)
  File "/home/<user>/.local/lib/python3.9/site-packages/StreamDeck/Transport/LibUSBHIDAPI.py", line 400, in write_feature
    return self.hidapi.send_feature_report(self.device_handle, payload)
  File "/home/<user>/.local/lib/python3.9/site-packages/StreamDeck/Transport/LibUSBHIDAPI.py", line 217, in send_feature_report
    raise TransportError("Failed to write feature report (%d)" % result)
StreamDeck.Transport.Transport.TransportError: Failed to write feature report (-1)
python3: ../../libusb/os/threads_posix.h:58: usbi_mutex_destroy: Assertion `pthread_mutex_destroy(mutex) == 0' failed.
Aborted (core dumped)
```

After the error I must unplug and re-plug the device USB port before it works again. Quitting from the Quit option in the system tray menu does close the app cleanly, and I can re-start it from systemd afterward.

When digging into the problem, I found this Stack Overflow answer detailing a QT "wakeup handler" that can handle the SIGINT signal sent by systemd (systemd will wait for a short time to let the app handle the signal before uncleanly terminating it).

https://stackoverflow.com/a/37229299/564755

I made a few tweaks since this app is built with PySide2, but this solution successfully allows me to stop the service from systemd and then restart it without needing to mess with the device. If you have any interest in accepting this change as a pull request, I can submit one.

Change: https://github.com/nemec/streamdeck-ui/commit/53042f3e3d2c531094d90c1f9e9eb8f8c500d0c6

**Comment by nemec** [comment]
Also, for the record, if anyone else is interested in a systemd install (I just learned about `systemd --user` but I _believe_ it will auto-start the app when you log in to your PC), here's what I did:

```shell
mkdir -p $HOME/.local/share/systemd/user/
nano $HOME/.local/share/systemd/user/streamdeck.service
```

Contents of service file. I'm using a virtualenv, but if you've installed with pip --user (the default instructions) you should be able to replace the execstart line with the output of `which streamdeck`, iirc.

```
[Unit]
Description=Streamdeck Controller
[Service]
Type=simple
TimeoutStartSec=0
WorkingDirectory=/home/<user>/prg/src/streamdeck-ui
ExecStart=/home/<user>/prg/src/streamdeck-ui/env/bin/python streamdeck_ui/gui.py -n
[Install]
WantedBy=default.target
```

```shell
systemctl --user daemon-reload
systemctl --user enable streamdeck.service
systemctl --user start streamdeck.service
```

To repro the issue I filed, follow the above systemd install steps then run:

```shell
systemctl --user stop streamdeck.service
systemctl --user start streamdeck.service
systemctl --user status streamdeck.service
```

Without my patch, the service should have failed with an error on the second start.

**Comment by fstamour** [comment]
I've had the same issue when trying this project out the first time. I ran `streamdeck` from the command line. Everything worked well, except the process wouldn't close when I sent it keyboard interrupts (`Ctrl+C`). 

The process finally closed when I called `kill` on it.

But now I have the `Failed to write feature report (-1)` error.

**Comment by dodgyrabbit** [comment]
@fstamour @nemec I've implemented a SIGINT and SIGTERM handler and tested that you can manage the service using `systemctl`.  It was committed in c0d2cd86b21ce3b8e6811fb24950045207bfba0a and I'll add it to the next release.

Thanks for the instructions on setting up the service in systemd. I'm going to add this to the documentation.

**Comment by dodgyrabbit** [comment]
Implemented in 2.0.14
